.github/workflows/test.yml

```diff
@@ -24,3 +24,32 @@ jobs:
     - name: Test with pytest
       run: |
         pytest -vv
+        
+  fedoratest:
+    name: fedora
+    runs-on: ubuntu-latest
```

Comment:
Can you run on fedora directly?

Reply:
No. GitHub only has Ubuntu, MacOS and Windows as base operating systems. Fedora offers a service [packit](https://packit.dev/), but getting a virtualized build is probably a good first step.